src/session.rs

```diff
@@ -105,9 +100,11 @@ pub fn start(config: &Config, initial_request: Option<String>) -> i32 {
                 match controllers.entry(route.clone()) {
                     Entry::Occupied(controller_entry) => {
                         if let Err(err) = controller_entry.get().worker.sender().send(request.clone())  {
-                            if let Some(fifo) = request.meta.fifo {
-                                cancel_blocking_request(fifo);
-                            }
+                            return_request_error(
+                                editor.to_editor.sender(),
+                                &request,
+                                "Language server is no longer running"
```

Comment:
> Display an error when lsp server is down
> 
> If an LSP server crashes, two requests fail in a row.

Yeah that's weird alright. We could try to restart automatically but maybe it's safer not to.
I'm testing with gopls (simulating crashes with `pkill gopls`) and I usually have only one failing request but sometimes two.

- sometimes the first lsp-hover after crash shows "Language server no longer running" and the second one works
- sometimes the first lsp-hover after crash shows nothing, the second "Language server no longer running" and the third one works

Reply:
I think I understand what's happening here now -- the language server transport layer takes a second to shut down. Upon a regular `pkill rust-analyzer` I get:

```
Oct 27 03:25:07.795 DEBG Language server closed pipe, stopping reading, module: kak_lsp::language_server_transport:134
Oct 27 03:25:07.796 INFO Waiting for Messages to language server to finish..., module: kak_lsp::thread_worker:19
Oct 27 03:25:07.796 DEBG Received signal to stop language server, closing pipe, module: kak_lsp::language_server_transport:193
Oct 27 03:25:07.796 DEBG Waiting for language server process end, module: kak_lsp::language_server_transport:92
```

If requests are sent during the one-second window it waits for process shutdown (?), they are ignored. Then, a second later it fully shuts down:
```
Oct 27 03:25:08.796 INFO ... Messages to language server terminated with ok, module: kak_lsp::thread_worker:21
Oct 27 03:25:08.796 INFO Waiting for Messages from language server to finish..., module: kak_lsp::thread_worker:19
Oct 27 03:25:08.796 INFO ... Messages from language server terminated with ok, module: kak_lsp::thread_worker:21
Oct 27 03:25:08.797 INFO Waiting for Language server errors to finish..., module: kak_lsp::thread_worker:19
Oct 27 03:25:08.797 INFO ... Language server errors terminated with ok, module: kak_lsp::thread_worker:21
```